mlir-tensorrt/compiler/include/mlir-tensorrt/Compiler/Client.h

```diff
@@ -145,8 +172,43 @@ class CompilerClient {
   /// used to create the PM.
   llvm::DenseMap<PassManagerKey, std::unique_ptr<CompilationTaskBase>>
       cachedPassManagers;
+
+  static llvm::ManagedStatic<llvm::StringMap<OptionsConstructorFuncT>> registry;
 };
 
+/// Helper to register option types with the client
+template <typename OptionsT, typename TaskT>
+StatusOr<std::unique_ptr<mlir::OptionsContext>>
+optionsCreateFromArgs(const CompilerClient &client,
+                      const std::vector<llvm::StringRef> &args) {
+  // Load available extensions.
+  mlir::MLIRContext *context = client.getContext();
+  mlir::plan::PlanDialect *planDialect =
+      context->getLoadedDialect<mlir::plan::PlanDialect>();
+  compiler::TaskExtensionRegistry extensions =
+      planDialect->extensionConstructors.getExtensionRegistryForTask<TaskT>();
+
+  auto result = std::make_unique<OptionsT>(std::move(extensions));
+
+  std::string err;
+  if (failed(result->parse(args, err))) {
+    std::string line = llvm::join(args, " ");
```

Comment:
The two lines in the body can be simplified to:

```
return getInternalErrorStatus("failed to parse options string \"{0:$[ ]}\" due to error {1}", llvm::iterator_range(args), err);
```

Reply:
I get an error with this:
> static void llvm::format_provider<llvm::StringRef>::format(const T &, llvm::raw_ostream &, StringRef) [T = llvm::StringRef, Enable = void]: Assertion `false && "Style is not a valid integer"' failed.

I can still merge the two lines though.